src/main/java/org/mastodon/mamut/nearest/NearestObjectStatModel.java

```diff
@@ -0,0 +1,472 @@
+					return false;
+
+			// Test stat only if we are not collecting number of neighbors.
+			if ( !( collectBy == CollectBy.MAX_DISTANCE && value == Value.DISTANCE_OR_N ) )
+				if ( o.stat != stat )
+					return false;
+
+			if ( o.include != include )
+				return false;
+
+			return true;
+		}
+
+		/**
+		 * Returns a function that map this stat specification to a double
+		 * value. The function is specified as <code>f(neighbor, center)</code>
+		 * where <code>neighbor</code> is one of the neighbors considered, and
+		 * <code>center</code> the spot which neighborhood is being collected.
+		 * 
+		 * @param featureModel
+		 *            a feature model from which to read feature values.
+		 * @return a new {@link ToDoubleBiFunction}.
+		 */
+		public ToDoubleBiFunction< Spot, Spot > eval( final FeatureModel featureModel )
+		{
+			switch ( value )
+			{
+			case DISTANCE_OR_N:
+				switch ( collectBy )
+				{
+				case MAX_DISTANCE:
+					return null;
+				case SPECIFY_N:
+					return ( s, center ) -> Util.distance( s, center );
+				default:
+					throw new IllegalArgumentException( "Unknown collection method: " + collectBy );
+				}
+			case FEATURE:
+			{
+				final FeatureProjection< Spot > projection = NearestObjectStatFeature.getProjection( this.featureID, featureModel );
+				return ( s, center ) -> projection.value( s );
+			}
+			default:
+				throw new IllegalArgumentException( "Unknown value definition: " + value );
+			}
+		}
+
+		public double summarize( final TDoubleArrayList arr )
+		{
+			if ( arr.isEmpty() )
+				return Double.NaN;
+			switch ( stat )
+			{
+			case MEAN:
+				return arr.sum() / arr.size();
+			case MEDIAN:
+				arr.sort();
+				return arr.getQuick( arr.size() / 2 );
+			case STD:
+				return Math.sqrt( variance( arr ) );
+			case MIN:
+				return arr.min();
+			case MAX:
+				return arr.max();
+			case SUM:
+				return arr.sum();
+			default:
+				throw new IllegalAddException( "This summarizing operation is unknown: " + stat );
+			}
+		}
+
+		public static Builder create()
+		{
+			return new Builder();
+		}
+	}
+
+	public enum Value
+	{
+		DISTANCE_OR_N( "Distance or N" ),
+		FEATURE( "Feature" );
+
+		private final String name;
+
+		Value( final String name )
+		{
+			this.name = name;
+		}
+
+		@Override
+		public String toString()
+		{
+			return name;
+		}
+	}
+
+	public enum Stat
+	{
+
+		MEAN( "Mean" ),
+		STD( "Std" ),
+		MIN( "Min" ),
+		MAX( "Max" ),
+		MEDIAN( "Median" ),
+		SUM( "Sum" );
+
+		private final String name;
+
+		Stat( final String name )
+		{
+			this.name = name;
+		}
+
+		@Override
+		public String toString()
+		{
+			return name;
+		}
+	}
+
+	public enum CollectBy
+	{
+
+		SPECIFY_N( "Specifying N" ),
+		MAX_DISTANCE( "Max distance" );
+
+		private final String name;
+
+		CollectBy( final String name )
+		{
+			this.name = name;
+		}
+
+		@Override
+		public String toString()
+		{
+			return name;
+		}
+	}
+
+	private static final double variance( final TDoubleArrayList arr )
```

Comment:
This private method is only called by the inner class NearestObjectStatItem. So it could be moved there.
The final modifier is not needed for a private method